src/test/java/test/DynamicGraphTest.java

```diff
@@ -74,15 +73,19 @@ public void test8() {
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
     dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
-
-    assertFreeNodesEquals(dg, a1, a2);
+    assertFreeNodesEquals(dg);
```

Comment:
I'm a bit nervous with the modification of this test, can you elaborate why `a1` and `a2` are no longer free now?